api/tests/core/offers/test_models.py

```diff
@@ -673,3 +674,29 @@ def test_can_activate_inactive_offer_if_venue_is_active_and_provider_is_not_null
         )
         factories.StockFactory(offer=offer_from_active_venue_provider)
         assert offer_from_active_venue_provider.isActivable
+
+
+class OfferfullAddressTest:
+    @pytest.mark.parametrize("has_label,has_street", [(True, True), (False, True), (True, False), (False, False)])
+    def test_full_address(self, has_label, has_street):
+        oa = offerers_factories.OffererAddressFactory(
+            label="label" if has_label else None,
+            address__street="street of the full address" if has_street else None,
+            address__postalCode="75000",
+            address__city="Paris",
+        )
+        offer = factories.OfferFactory(offererAddress=oa)
+        assert not offer.fullAddress.startswith("-")
+        assert not offer.fullAddress.endswith("-")
+        assert not offer.fullAddress.startswith(" ")
+        assert not offer.fullAddress.endswith(" ")
+        if has_label:
+            if has_street:
+                offer.fullAddress == "label - street of the full address - 75000 - Paris"
+            else:
+                offer.fullAddress == "label - 75000 - Paris"
+        else:
+            if has_street:
+                assert offer.fullAddress == "street of the full address - 75000 - Paris"
+            else:
+                assert offer.fullAddress == "75000 - Paris"
```

Comment:
Suggestion : la même chose plus clairement : 
```suggestion
class OfferfullAddressTest:
    @pytest.mark.parametrize("label,street,expected_full_address", [
        ("label", "street of the full address", "label - street of the full address - 75000 - Paris"),
        (None, "street of the full address", "street of the full address - 75000 - Paris"),
        ("label", None, "label - 75000 - Paris"),
        (None, None, "75000 - Paris"),
    ])
    def test_full_address(self, label, street, expected_full_address):
        oa = offerers_factories.OffererAddressFactory(
            label=label,
            address__street=street,
            address__postalCode="75000",
            address__city="Paris",
        )
        offer = factories.OfferFactory(offererAddress=oa)
        assert offer.fullAddress == expected_full_address
```